gfw-api/gfw-ogr-api
# Feat/add v2 mapshaper

## Aim

We are trying to create a v2 of the `/convert` endpoint. This will replace the current `convert` service active on GFW (that is used to receive uploaded shapes, and returns a valid geojson, ready to be posted to the Geostore service).

The updated service will take a couple of optional params `simplify` (which should have an integer value from 0-100), and clean (boolean). After ogr2ogr reads the file and return a geojson, the geojson will be passed to mapshaper with the params. The mapshaper response should be serialized and returned.

## Steps to test:
Run control tower, and then run this service.

You can send a query such as:

```
curl --request POST \
  --url 'http://localhost:9000/v2/ogr/convert?simplify=60&clean=true' \
  --header 'Accept: application/json, text/pain, */*' \
  --header 'Accept-Encoding: gzip, deflate' \
  --header 'Cache-Control: no-cache' \
  --header 'Connection: keep-alive' \
  --header 'Content-Length: 701' \
  --header 'Content-Type: multipart/form-data; boundary=--------------------------726770954131108734947446' \
  --header 'Host: localhost:9000' \
  --header 'Postman-Token: 14af6ef0-c62e-4181-bf53-56de3f5db9f6,998e5d06-8af6-4f23-8dde-fda246565825' \
  --header 'User-Agent: PostmanRuntime/7.17.1' \
  --header 'cache-control: no-cache' \
  --header 'content-type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW' \
  --form file=@/Users/Ben/Desktop/test-shape.geojson
```
Note that for testing the` test-shape.geojson` is held in the following zip file. You will also need to change the path to the file in the `--form-file` line above.

[test-shape.geojson.zip](https://github.com/gfw-api/gfw-ogr-api/files/3735253/test-shape.geojson.zip)

## Current issue 

At the moment our wonderful update is not working. Although the V2 route is present, and the code is being executed, It seems like there is an issue with mapshaper not returning an object. I am guessing it is that @01painadam and I have currently done something wrong. The issue may be that mapshaper is not returning its promise, but we could use some help to solve this one.
